compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmStaticAnnotationLowering.kt

```diff
@@ -60,70 +63,59 @@ private class CompanionObjectJvmStaticLowering(val context: JvmBackendContext) :
 
         companion?.declarations?.filter(::isJvmStaticFunction)?.forEach { declaration ->
             val jvmStaticFunction = declaration as IrSimpleFunction
-            val proxy = createProxy(jvmStaticFunction, irClass, companion)
-            irClass.addMember(proxy)
+            if (jvmStaticFunction.isExternal) {
+                // We move external functions to the enclosing class and potentially add accessors there.
+                // The JVM backend also adds accessors in the companion object, but these are superfluous.
+                val staticExternal = irClass.addFunction {
+                    updateFrom(jvmStaticFunction)
+                    name = jvmStaticFunction.name
+                    returnType = jvmStaticFunction.returnType
+                }.apply {
+                    copyTypeParametersFrom(jvmStaticFunction)
+                    extensionReceiverParameter = jvmStaticFunction.extensionReceiverParameter?.copyTo(this)
+                    valueParameters = jvmStaticFunction.valueParameters.map { it.copyTo(this) }
+                    annotations = jvmStaticFunction.annotations.map { it.deepCopyWithSymbols() }
+                }
+                companion.declarations.remove(jvmStaticFunction)
+                companion.addProxy(staticExternal, companion, isStatic = false)
+            } else {
+                irClass.addProxy(jvmStaticFunction, companion)
+            }
         }
-
     }
 
-    private fun createProxy(target: IrSimpleFunction, irClass: IrClass, companion: IrClass): IrSimpleFunction {
-        return buildFun {
-            updateFrom(target)
+    private fun IrClass.addProxy(target: IrSimpleFunction, companion: IrClass, isStatic: Boolean = true) =
+        addFunction {
```

Comment:
Did you omit `updateFrom` here intentionally? It looks like that might cause some flags from the declaration not propagated to the proxy, such as `isOperator`, `isSuspend`, etc. I'm not totally sure what's the correct behavior here

Reply:
Yes. The previous code manually set all of the other flags to `False`. Apart from `isOperator`, the two implementations end up doing the same thing and I honestly don't know what we should be doing about most of these flags.

As far as I can tell we write the metadata based on the class descriptors, so I'm not sure whether these flags even end up in the output.